app/controllers/host_controller.rb

```diff
@@ -6,96 +6,45 @@ class HostController < ApplicationController
 
   include Mixins::GenericSessionMixin
   include Mixins::GenericListMixin
+  include Mixins::GenericShowMixin
   include Mixins::MoreShowActions
 
-  def show
-    return if perfmenu_click?
+  def self.display_methods
+    %w(hv_info os_info devices network storage_adapters performance timeline storages resource_pools vms miq_templates compliance_history)
+  end
 
-    @lastaction = "show"
+  def display_config_info
     @showtype = "config"
+    title = case @display
+            when "hv_info" then _("VM Monitor Information")
+            when "os_info" then _("OS Information")
+            when "devices" then _("Devices")
+            end
+    drop_breadcrumb(:name => "#{@record.name} (#{title})",
+                    :url  => "/host/show/#{@record.id}?display=#{@display}")
```

Comment:
In `app/controllers/mixins/generic_show_mixin.rb` we have:

```
    def show_link(record, options = {})                                                             
```

That method deals with `restful?` so that we will not have to revisit code that uses it once we convert all the controllers to restfuf urls. Please, use that.